### PR TITLE
make warnings from the validator available rather then ignoring them

### PR DESCRIPTION
**Pull request by dirk-thomas**
If e.g. `xmllint` fails to access a referenced xml-model file the warning was not visible to the user. If the validation fails the remaining output it always shown in a message.

### CODE REVIEW

File: lib/init.coffee [hunk 1]
```diff
@@ -211,10 +211,18 @@ module.exports =
...
+          for message in messages
+            message.type = 'Error'
+            message.text = message.text + ' (' + schemaUrl + ')'
+            message.filePath = textEditor.getPath()
+        else if output.indexOf('- validates') is -1
+          messages.push({
+            type: 'Error'
+            text: output
+            filePath: textEditor.getPath()
+            range: [[0, 0], [0, 0]]
```
Comment (Arcanemagus) [inline lib/init.coffee]: If you omit this range it is still a valid message, linter will show it as associated with the file but not give it a range within the file.

Comment (dirk-thomas) [inline lib/init.coffee]: That's why I always keep my own PRs open for a day or two. You always have a good comment :wink: Thanks - updated in c9e5bf81504543deb4196194999301ece6f45673

File: spec/linter-xmllint-spec.coffee [hunk 2]
```diff
@@ -114,6 +114,19 @@ describe 'The xmllint provider for Linter', ->
           expect(messages.length).toEqual 1
           expect(messages[0].range).toEqual [[3, 2], [3, 21]]
     waitsForPromise ->
+      return atom.workspace.open(__dirname + '/fixtures/invalid/xml-model-unavailable.xml').then (editor) ->
+        return lint(editor).then (messages) ->
+          expect(messages.length).toBeGreaterThan 0
+          expect(messages.length).toBeLessThan 3
```
Comment (dirk-thomas) [inline spec/linter-xmllint-spec.coffee]: @Arcanemagus Another question regarding Coffeescript: is there a better way to express this - I don't like the fact that it mentions `0` and `3` while the actual interval is `[1, 2]`? I would imagine something like `.toBeIn [1, 2]` or at least `.toBeGreaterOrEqualThen 1` and `.toBeLessOrEqualThan 2` but couldn't find anything.

Comment (Arcanemagus) [inline spec/linter-xmllint-spec.coffee]: Unfortunately I don't see anything that fits better in [the list of matchers](http://jasmine.github.io/1.3/introduction.html#section-Included_Matchers).

Comment (dirk-thomas) [inline spec/linter-xmllint-spec.coffee]: Thanks for the comment. I wasn't sure if I was missing anything.
